### ISSUE TITLE
templates.version is incremented but never pinned; template structure is not snapshotted while the rating system is

### ISSUE THREAD

**Issue by important-new** [InspectorHub/OpenInspection#292]
`templates.version` is incremented on every update and read by nothing. Meanwhile the rating system attached to the same inspection **is** frozen, with a comment explaining exactly why. Template structure is not.

## Evidence

`server/services/template.service.ts:243` — every `updateTemplate` writes `version: existing.version + 1`.

The only reads of `templates.version` in the repo are display: the list query at `template.service.ts:92` and the row it returns at `:109`. Nothing pins a version, and no inspection stores one.

Contrast `server/services/inspection/inspection-results.service.ts:193`, on the path that seeds an inspection's results row:

> Sprint 2 S2-1 — when seeding an `inspection_results` row for the first time, also freeze the active rating system onto the row so future edits to the source system never mutate this report.

So `ratingSystemSnapshot` is written and the rating system is safe. The template schema that defines the sections and items is read live every time.

## Failure

An inspector opens an inspection. Someone else (or the same person, from the templates page) edits the template — removes a section, renames an item, reorders. The open inspection's structure changes underneath, and there is no version boundary that could tell you it happened.

Published reports are not at risk: they render from their own snapshot. The exposure is the in-progress window, which for a multi-day job (radon retrieval, sewer scope on a different day) is not short.

## What to decide

Two coherent options, and they are not the same amount of work:

1. **Snapshot the template schema onto the inspection**, the way the rating system already is. Consistent with the existing pattern; costs storage per inspection, and a tenant fixing a typo in a template gets no benefit on open jobs.
2. **Pin `templateId` + `version`** and keep versions immutable — an edit writes a new row rather than mutating. Cheaper per inspection, but changes what "delete a template" means and needs a retention story for old versions.

Option 1 matches what the codebase already does elsewhere and is the smaller change. Option 2 is the more correct model if template history ever becomes a product feature.

Either way, `version` should stop being incremented into a void.

**Comment by important-new** [comment]
## Competitor evidence — checked before choosing, 2026-08-02

Two competitors' captured API payloads were searched for any template-version or
lock concept.

**Spectora — no template version exists anywhere.** Their template payloads
(`sections.json`, `items.json`, `item_attributes.json`, `template_ratings.json`)
carry `template_id` and no version field of any kind. What their inspection
carries instead is a build state:

```
reports_present: true
reports_built:   true
reports_awaiting_review: false
reports_published: false
```

Combined with their own documentation — *"the report will **automatically build
from that template** whenever that service is scheduled"* — this is a
**materialisation** model: the template produces the report at build time, and
the report is thereafter its own document. A later template edit has nothing to
reach into.

**Inspector Toolbelt — `templateVersion` exists, but not for this.** It appears
only on a marketplace record, next to `isPublished: true`, `parentId` and
`authorId`. That is a **distribution** version for shared/published templates,
not a pin on an inspection. Their `itb_inspections.json` carries `templateId`
with no version.

So the mainstream pattern is:

- **materialise at build time** — mid-inspection template edits do not reach a
  built report, because there is no live resolution to reach through
- **version numbers are for distributing templates**, not for pinning inspections

## Decision

**Option 1 — snapshot the template structure onto the inspection.** It matches
both competitors' effective behaviour and, more importantly, it matches what this
codebase already does one line away: `inspection-results.service.ts:193` freezes
the rating system onto the results row *"so future edits to the source system
never mutate this report"*. Template structure is the same class of thing and was
simply missed.

The consequence stated in the issue stands and is accepted: a tenant fixing a
typo in a template gets no benefit on already-open jobs. Both competitors accept
the same trade.

`templates.version` stops being incremented into a void. If it is kept, it is
kept for the Inspector Toolbelt use — distributing a template — and that is a
different feature with its own issue, not a silent side effect of saving an edit.

**Comment by important-new** [comment]
Status audit against the tree, 2026-08-10 — half of this is no longer true.

**"template structure is not snapshotted": stale.** `inspections.template_snapshot` (json) and `inspections.template_snapshot_version` exist in `server/lib/db/schema/inspection/core.ts:105-106`, and they are written — `inspection-results.service.ts` sets `templateSnapshot` on the result-write paths (lines 246 and 320), `inspection-reinspection.service.ts` carries the baseline's snapshot forward, and `inspection-core.service.ts` seeds it. So the structure IS pinned per inspection.

**"templates.version is incremented but never pinned": still true.** There is no template-version column on `inspections` — grep for `templateVersion`/`template_version` in `schema/inspection/core.ts` returns nothing. The snapshot records the shape; nothing records which version number produced it, so an audit cannot say "this inspection was taken on template v4" without diffing the snapshot against template history.

Not closing: the remaining half is the one the title leads with. Suggest the scope is now narrower than written — pin the version number alongside the snapshot that already exists, rather than build snapshotting.

**Comment by important-new** [comment]
Closing — verified against `main` (`146560b4`). Both halves of this are now in the tree.

**The template structure is snapshotted onto the inspection**, which was option 1 in the issue body. Every creation path writes it:

- `server/services/inspection/inspection-core.service.ts:205-213` — `templateSnapshot = tpl.schema`, `templateSnapshotVersion = tpl.version`
- `server/services/inspection-request.service.ts:170-171` and `:263-264` — same, on both request-conversion paths
- `server/services/inspection/inspection-reinspection.service.ts:148-149` — copies the baseline's snapshot rather than re-reading the template

**The editor reads the snapshot, not the live template.** `app/routes/inspection-edit/loader.server.ts:62` states it as a requirement in the source: *"The base structure MUST come from the inspection's OWN templateSnapshot"*, and `:72` resolves from the inspection row. So the failure this issue describes — someone edits the template and an open inspection's structure changes underneath — no longer reproduces.

**`templates.version` is no longer incremented into a void.** `POST /api/inspections/{id}/template/upgrade` (`server/api/inspection-sync.ts:196-215`) compares the inspection's pinned `templateSnapshotVersion` against the master `templates.version`, no-ops when the master has not moved, and re-snapshots with an audit entry (`inspection.template_upgraded`) when it has. Pulling template changes into an open job is now an explicit, recorded action rather than something that happens by itself.

Per-inspection structural tailoring (`updateTemplateSnapshot`, `inspection-results.service.ts:239-248`) deliberately does not bump the pinned version — it edits this inspection's own copy and never touches the source template row.
